# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/mediasink/AudioSink.h
# Commit: 4076e2055960
# Full Hash: 4076e2055960a3e461f31839564bfdb5f1fd402c
# Author: Paul Adenot <paul@paul.cx>
# Date: 2022-05-10 03:50:31
# Regressor Bug: 1743834
# File Overlap Count: 1
# Description:
#   Bug 1743834 - Correctly set the start time when the audio callbacks actually are called, the first time audio audio clock starts to be in use. r=alwu,media-playback-reviewers
#   
#   The base clock for the new audio stream is precisely the time when the clock
#   source changes from the system clock to the audio clock, accounting for the time
#   it takes to actually starts the stream. This allows avoiding discontinuities
# ==============================================================================

diff -r 966ceb9947fa -r 4076e2055960 dom/media/mediasink/AudioSink.h
--- a/dom/media/mediasink/AudioSink.h	Mon May 09 13:20:50 2022 +0000
+++ b/dom/media/mediasink/AudioSink.h	Mon May 09 13:20:51 2022 +0000
@@ -83,6 +83,10 @@
     return mAudioStream && mAudioStream->CallbackStarted();
   }
 
+  void UpdateStartTime(const media::TimeUnit& aStartTime) {
+    mStartTime = aStartTime;
+  }
+
  private:
   // Allocate and initialize mAudioStream. Returns NS_OK on success.
   nsresult InitializeAudioStream(const PlaybackParams& aParams);
@@ -112,7 +116,7 @@
   // The presentation time of the first audio frame that was played.
   // We can add this to the audio stream position to determine
   // the current audio time.
-  const media::TimeUnit mStartTime;
+  media::TimeUnit mStartTime;
 
   // Keep the last good position returned from the audio stream. Used to ensure
   // position returned by GetPosition() is mono-increasing in spite of audio